pygti/schemas.py

```diff
@@ -84,7 +95,74 @@
         "filter": [FilterEntry],
         "serviceTypes": [FilterServiceType],
         "useRealtime": bool,
+    },
+)
+
+SIRequest = Schema.extend(BaseRequestType, {"station": SDName})
+
+ContSearchByServiceId = Schema(
+    {
+        "serviceId": int,
+        "lineKey": str,
+        "plannedDepArrTime": GTITime,
+        "additionalOffset": int,
     }
 )
 
-SIRequest = Schema({"station": SDName})
+# kinds: GTI Docs say int but traffic on the website uses strings
+# Same thing with the value of the penalty. All are strings
+TariffInfoSelector = Schema({"tariff": str, "tariffRegions": bool, "kinds": [str],})
+PenaltyName = In(
+    [
+        "ChangeEvent",
+        "ExtraFare",
+        "Walker",
+        "AnyHandicap",
+        "ToStartStationBy",
+        "TimeRange",
+        "ForVisitors",
+        "desiredType",
+        "DesiredLine",
+    ]
+)
+
+Penalty = Schema({"name": PenaltyName, "value": str})
```

Comment:
See "2.3.3 Penalties", allowed values depend on the PenaltyName, so both string and int should be allowed here

Reply:
I thought so, too.
Tests told me that they are all strings.
Sending integers will give you an error.
They will interpret the strings as integers i guess but you have to send strings

Reply:
See payloadsForTesting.py l.281ff.
The example was copied from a request made by the hvv webpage